### PR TITLE
Test plan 4.15.3. Atomic references

### PR DESCRIPTION
**Pull request by bibrakc**
This pull request aims at test plan for 4.15.3. Atomic references.

### CODE REVIEW

File: test_plans/atomic_ref.asciidoc [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+
+* `memory_scope::work_item`
+* `memory_scope::sub_group`
+* `memory_scope::work_group`
+* `memory_scope::device`
+* `memory_scope::system`
+
+And For 64-bit data types on devices that have `aspect::atomic64`.
+
+Check if atomically replaces the value of the object referenced by this atomic_ref with value operand and returns the original value of the referenced object.
```
Comment (AJcodes) [inline test_plans/atomic_ref.asciidoc:176]: To review

File: test_plans/atomic_ref.asciidoc [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+This is a test plan for the APIs described in SYCL 2020 section 4.15.3. Atomic references
+
+Estimated development time is three days.
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
```
[the remark this answers is not in the document]
Comment (gmlueck) [reply test_plans/atomic_ref.asciidoc:15]: Just FYI ... the spec should be clarified w.r.t. using `atomic_ref` on the host.  I've captured some of the open issues here (requires Khronos access):

https://gitlab.khronos.org/sycl/Specification/-/issues/637

File: test_plans/atomic_ref.asciidoc [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+
+* `access::address_space::global_space`
+* `access::address_space::local_space`
+* `access::address_space::private_space`
+* `access::address_space::generic_space`
+
+Call constructors:
+
+* `atomic_ref(T&)`
+Check if the values of newly created object match with defult values and the stored value passed as `T&`?
```
Comment (keryell) [inline test_plans/atomic_ref.asciidoc:91]: Use a spell-checker
```suggestion
Check if the values of newly created object match with default values and the stored value passed as `T&`?
```

File: test_plans/atomic_ref.asciidoc [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+* `memory_scope::sub_group`
+* `memory_scope::work_group`
+* `memory_scope::device`
+* `memory_scope::system`
+
+And For 64-bit data types on devices that have `aspect::atomic64`.
+
+For equal values: check if it replaces the value of the referenced object with the value of `desired`.
+
+For uneual values: check if it assigns the original value of the referenced object to `expected`.
```
Comment (keryell) [inline test_plans/atomic_ref.asciidoc:234]:
```suggestion
For unequal values: check if it assigns the original value of the referenced object to `expected`.
```

**Comment by bader** [comment]
@bibrakc, I think https://github.com/KhronosGroup/SYCL-CTS/pull/450 supersedes this PR. If so, can we close this one?

**Pull request closed without merging**